### PR TITLE
Add description for removing holidays by name

### PR DESCRIPTION
**Pull request by matthewgottlieb**
## Proposed change

The workday integration loads region-specific holidays.  Some of these holidays may not be days off from work, and it is tedious to remove each of these holidays by date, year after year.  The Python module [holidays](https://pypi.org/project/holidays/) supports removal by partial string match, the parent PR exposes that functionality to the integration.  This PR documents the new functionality.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/52700
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Review by frenck — APPROVED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

File: source/_integrations/workday.markdown [hunk 1]
```diff
@@ -66,7 +66,7 @@ add_holidays:
   required: false
   type: list
 remove_holidays:
-  description: "Remove holidays (treat holiday as workday). Needs to formatted as `YYYY-MM-DD`."
+  description: "Remove holidays (treat holiday as workday). Can be formatted as `YYYY-MM-DD` or by name for a partial string match (e.g. Thanksgiving)."
```
Comment (TomBrien) [inline source/_integrations/workday.markdown:69]: We should point users to where to find the correctly formatted strings to use. There is no nicely maintained list so I think the best compromise is to point to [the source](https://github.com/dr-prodigy/python-holidays/tree/master/holidays/countries) where they can look in the `.py` file for their country.

**Review by TomBrien — CHANGES_REQUESTED** [review]

File: source/_integrations/workday.markdown (hunk 1 above)
Comment (frenck) [reply source/_integrations/workday.markdown:69]: I don't agree on sending our end-users to Python source files.

Comment (TomBrien) [reply source/_integrations/workday.markdown:69]: ~What do you suggest as an alternate. `Holidays` is super picky on format and the library doesn't offer a clear list anywhere I could find~

So I think this is the upstream source for the holidays: https://en.wikipedia.org/wiki/List_of_holidays_by_country so perhaps that?

Comment (TomBrien) [reply source/_integrations/workday.markdown:69]: Although I note that for example King's Day in the Netherlands is not translated in the Wikipedia page (presumably is in the Dutch version) but is in `Holidays`

Comment (matthewgottlieb) [reply source/_integrations/workday.markdown:69]: I have to agree that directing users to Python source files isn't the most elegant.  I'd thought the partial string match would be forgiving enough for the majority of cases and debug logs would be available for the rest, but a more accessible reference would be ideal.

I'd planned a future PR (once I got the hang of contributing) to add additional attributes to the workday integration, potentially including the list of "known holidays".  Would this be better?

Comment (TomBrien) [reply source/_integrations/workday.markdown:69]: That sounds good although not sure attributes is the way to do it. Migrating to a config flow and allowing a picker in the UI seems like the best solution long term. Looking back I think the partial matching is going to work well enough and this is already more robust than date-format removal

**Review by TomBrien — APPROVED** [review]

**Pull request merged**